### PR TITLE
Angular SDK

### PR DESCRIPTION
**Pull request by pablotransifex**
Angular SDK implementation for Transifex Native.

Includes:
- Components: T, UT, LanguagePicker
- Decorator: T
- Service: TranslationService

CLI:
- Parsers: HTML template for T, UT tags (file) and Decorator in .ts files

### CODE REVIEW

File: packages/angular/projects/tx-native-angular-sdk/src/lib/interfaces.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+export interface ITranslationServiceConfig {
+  token: string;
+  secret: string;
+  sourceLocale: string;
+}
```
Comment (nbasili) [inline packages/angular/projects/tx-native-angular-sdk/src/lib/interfaces.ts:5]: You may check https://github.com/transifex/transifex-javascript/tree/master/packages/native#initialize-library for initialization options on the native SDK. For example sourceLocale is no longer valid and there are more options there.

File: packages/angular/projects/tx-native-angular-sdk/src/lib/Services/translation.service.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+   * Initializes the translation service
+   * @param {ITranslationServiceConfig} config
+   * @returns void
+   */
+  public async init(config: ITranslationServiceConfig) {
+    tx.init({
+      token: config.token,
+      secret: config.secret,
+      sourceLocale: config.sourceLocale,
+    });
```
Comment (nbasili) [inline packages/angular/projects/tx-native-angular-sdk/src/lib/Services/translation.service.ts:38]: Any reason to not just pass `tx.init(config)` here?

File: packages/angular/projects/tx-native-angular-sdk/src/lib/Services/translation.service.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    this._localeChanged.next(locale);
+  }
+
+  /**
+   * Gets the current locale
+   * @returns string
+   */
+  public getLocale(): string {
+    return tx.currentLocale || '';
+  }
```
Comment (nbasili) [inline packages/angular/projects/tx-native-angular-sdk/src/lib/Services/translation.service.ts:58]: Maybe `tx.getCurrentLocale()` is better

File: packages/angular/projects/tx-native-angular-sdk/src/lib/Services/translation.service.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+  /**
+   * Translate a string
+   * @param {string} str
+   * @param {Object} params
+   * @returns void
+   */
+  public translate(str: string, params: Object): string {
+    this._contentTranslated.next(tx.currentLocale || '');
```
Comment (nbasili) [inline packages/angular/projects/tx-native-angular-sdk/src/lib/Services/translation.service.ts:75]: `tx.getCurrentLocale()`

File: packages/angular/tslint.json [hunk 5]
```diff
@@ -0,0 +1,140 @@
+{
+  "extends": "tslint:recommended",
```
Comment (nbasili) [inline packages/angular/tslint.json:2]: Could we use an `airbnb` compatible linter for that?

File: packages/cli/src/commands/push.js [hunk 6]
```diff
@@ -37,7 +37,7 @@ class PushCommand extends Command {
       : path.join(pwd, args.pattern);
 
     if (isFolder(filePattern)) {
-      filePattern = path.join(filePattern, '**/*.{js,jsx,ts,tsx}');
+      filePattern = path.join(filePattern, '**/*.{js,jsx,ts,tsx,html}');
```
Comment (nbasili) [inline packages/cli/src/commands/push.js:40]: You also need to update the README and the CLI help text to include `html` as well.

**Review by nbasili — CHANGES_REQUESTED** [review]

File: packages/angular/projects/tx-native-angular-sdk/src/lib/interfaces.ts (hunk 1 above)
Comment (pablotransifex) [reply packages/angular/projects/tx-native-angular-sdk/src/lib/interfaces.ts:11]: Done

File: packages/angular/projects/tx-native-angular-sdk/src/lib/Services/translation.service.ts (hunk 2 above)
Comment (pablotransifex) [reply packages/angular/projects/tx-native-angular-sdk/src/lib/Services/translation.service.ts]: Done

File: packages/angular/projects/tx-native-angular-sdk/src/lib/Services/translation.service.ts (hunk 3 above)
Comment (pablotransifex) [reply packages/angular/projects/tx-native-angular-sdk/src/lib/Services/translation.service.ts:58]: Done

File: packages/angular/projects/tx-native-angular-sdk/src/lib/Services/translation.service.ts (hunk 4 above)
Comment (pablotransifex) [reply packages/angular/projects/tx-native-angular-sdk/src/lib/Services/translation.service.ts]: Done

File: packages/angular/tslint.json (hunk 5 above)
Comment (pablotransifex) [reply packages/angular/tslint.json:2]: I've migrated TSLint to ESLint, next step, try to use airbnb

Comment (pablotransifex) [reply packages/angular/tslint.json:2]: Added also AirBnB, not sure if it's the best, but there is a try :)

File: packages/cli/src/commands/push.js (hunk 6 above)
Comment (pablotransifex) [reply packages/cli/src/commands/push.js:40]: done

File: packages/angular/package.json [hunk 7]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "@transifex/angular",
+  "name": "@transifex/angular-projects",
```
Comment (pablotransifex) [inline packages/angular/package.json:2]: Here I had this for name: tx-native-angular-sdk, maybe we can use the name of the folder for this?

**Review by pablotransifex — COMMENTED** [review]
A small comment

**Review by nbasili — APPROVED** [review]

**Pull request merged**